doc: Release Notes for 2_9_0-nRF54H20 (#19597)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-2.9.0-nrf54h20-rc1.rst

* :ref:`nrf_auraconfig` sample:

  * Fixed the data sending (OCT-3251).
    Data is now sent on all BISes when generated by the application (no SD card inserted).

Edge Impulse samples
--------------------

* Added support for the :ref:`zephyr:nrf54h20dk_nrf54h20` board target in the following samples:

  * :ref:`ei_data_forwarder_sample`
  * :ref:`ei_wrapper_sample`

Matter samples
--------------

* Updated:

  * All Matter samples that support low-power mode to enable the :ref:`lib_ram_pwrdn` feature.
    It is enabled by default for the release configuration of the following samples:

    * :ref:`matter_lock_sample`
    * :ref:`matter_light_switch_sample`
    * :ref:`matter_smoke_co_alarm_sample`
    * :ref:`matter_window_covering_sample`

  * All Matter samples to enable the ZMS file subsystem in all devices that contain MRAM, such as the nRF54H Series devices.

* Disabled pausing Matter watchdog while CPU is in idle state in all Matter samples.
  To enable it, set the :ref:`CONFIG_NCS_SAMPLE_MATTER_WATCHDOG_PAUSE_IN_SLEEP<CONFIG_NCS_SAMPLE_MATTER_WATCHDOG_PAUSE_IN_SLEEP>` Kconfig option to ``y``.

* :ref:`matter_smoke_co_alarm_sample` sample:

>   * Added:
> 
>     * Support for ICD dynamic SIT LIT switching (DSLS).

Reviewer: ```suggestion
  * Added support for ICD dynamic SIT LIT switching (DSLS).
```

  * Drivers adapted for the following peripherals:

    * UARTE120
    * PWM120
    * SPIM120
    * EXMIF

  * UICR validation support.
  * MSPI EXMIF support.

Added the following features as experimental:

* Bluetooth:

  * :ref:`Parallel scanning and initiating connection <bt_scanning_while_connecting>`.
    It reduces the overall waiting time when there are several Bluetooth devices to be discovered and connected at the same time.

Improved the following features:

* Better BICR handling and generation.
* Enhanced power management features across various subsystems.

Implemented the following bug fixes:

* NCSDK-30802: Resolved an issue where the nRF54H20 device suddenly stopped transmitting ESB packets after nrfxlib 3.9.0.
* NCSDK-30161: Fixed an assertion during boot time caused by a combination of :kconfig:option:`CONFIG_ASSERT`, :kconfig:option:`CONFIG_SOC_NRF54H20_GPD`, and external flash.
* NCSDK-30117: Ensured that declaring a MEM component pointing to a memory region not assigned to a particular core is no longer possible.
* NCSDK-29682: Added support for the `cose-alg-sha-512` algorithm in the SUIT module.

Limitations
***********

> On the nRF54H20 SoC, the device firmware update (DFU) procedure from external flash memory does not work with the new flash memory driver based on the MSPI EXMIF handling.

Reviewer 2: ```suggestion
On the nRF54H20 SoC, the Device Firmware Update (DFU) procedure from external flash memory does not work with the new flash memory driver based on the MSPI EXMIF handling.
```

    You can use this to limit the slow poll interval for the ICD LIT device while it is temporarily working in the SIT mode.
  * New device types:

    * Water heater
    * Solar power
    * Battery storage
    * Heat pump
    * Mounted on/off control
    * Mounted dimmable load control

* Updated:

  * Thermostat cluster with support for scheduling and preset modes, like vacation, and home or away settings.
  * Electric Vehicle Supply Equipment (EVSE) with support for user-defined charging preferences, like specifying the time when the car will be charged.
  * Occupancy sensing cluster with features like radar, vision, and ambient sensing.
  * Intermittently Connected Devices feature with enhancements for the Long Idle Time (LIT) and Check-In protocol.
    With these enhancements, the state of this feature is changed from provisional to certifiable.

Thread
------

* Added Kconfig options for configuring the MLE child update timeout, child supervision interval, and child supervision check timeout.

Zigbee
------

* Updated:

>   * :ref:`nrfxlib:zboss` to v3.11.7.0 and platform v5.1.7 (``v3.11.6.0+5.1.7``).

Reviewer 2: ```suggestion
  * :ref:`nrfxlib:zboss` to v3.11.6.0 and platform v5.1.7 (``v3.11.6.0+5.1.7``).
```
This was changed in the 2.9 release notes - https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/releases_and_maturity/releases/release-notes-2.9.0.html#zigbee

    They contain several fixes related to malfunctioning in a heavy traffic environment and more.
    For details, see :ref:`zboss_changelog`.
  * The :ref:`ZBOSS Network Co-processor Host <ug_zigbee_tools_ncp_host>` package to the new version v2.2.5.

Applications
============

This section provides detailed lists of changes by :ref:`application <applications>`.

Machine learning
----------------

* Updated the application to enable the :ref:`Zephyr Memory Storage (ZMS) <zephyr:zms_api>` file system for the :ref:`zephyr:nrf54h20dk_nrf54h20` board.

IPC radio firmware
------------------

* Updated the application to enable the :ref:`Zephyr Memory Storage (ZMS) <zephyr:zms_api>` file system in all devices that contain MRAM, such as the nRF54H Series devices.

Connectivity bridge
-------------------

* Updated the handling of USB CDC ACM baud rate requests to make sure the baud rate is set correctly when the host requests a change.

>   This fixes an issue when using GNU screen with the Thingy:91 X.

Reviewer 2: Shouldn't this be removed as it is not part of the nRF54H20?

    * Use non-zero Bluetooth LE peripheral latency while USB is suspended.
      This is done to prevent peripheral latency increase requests from :ref:`nrf_desktop_ble_latency` on peripheral's end.
    * Revert the USB suspended Bluetooth LE connection parameter update when USB cable is disconnected.

  * The :ref:`nrf_desktop_ble_scan` to always use a connection interval of 10 ms for peripherals without Low Latency Packet Mode (LLPM) support if a dongle supports LLPM and more than one Bluetooth LE connection.
    This is required to avoid Bluetooth Link Layer scheduling conflicts that could lead to HID report rate drop.

* Removed imply for partial erase feature of the nRF SoC flash driver (:kconfig:option:`CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE`) for the USB next stack (:ref:`CONFIG_DESKTOP_USB_STACK_NEXT <config_desktop_app_options>`).
  The partial erase feature was used as a workaround for device errors that might be reported by the Windows USB host in Device Manager if a USB cable is connected while erasing a secondary image slot in the background.
  The workaround is no longer needed after the nRF UDC driver was improved.

> 

Reviewer 3: SUIT: Recovery application

Updated:
Added support for trigger entering recovery by pressing HW button or calling dedicated API. More information is here https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/app_dev/device_guides/nrf54h/ug_nrf54h20_suit_recovery.html

.. _ncs_release_notes_290-nrf54h20-rc1:

|NCS| v2.9.0-nRF54H20-rc1 Release Notes

> #######################################

Reviewer 2: I think the following pages need to be updated with the "v2.9.0-nRF54H20-rc1" version:
1. doc/nrf/dev_model_and_contributions/adding_code.rst
2. doc/nrf/shortcuts.txt - If the toolchain manager is also updated.

.. contents::
   :local:
   :depth: 2

|NCS| delivers reference software and supporting libraries for developing low-power wireless applications with Nordic Semiconductor products.
The SDK includes open source projects (TF-M, MCUboot, OpenThread, Matter, and the Zephyr RTOS), which are continuously integrated and redistributed with the SDK.

The |NCS| v2.9.0-nrf54h20-rc1 is a nRF54H20-exclusive release tag, only supporting products based on the nRF54H20 SoC.

Release notes might refer to "experimental" support for features, which indicates that the feature is incomplete in functionality or verification, and can be expected to change in future releases.
To learn more, see :ref:`software_maturity`.

Highlights
**********

Added the following features as supported:

* nRF54H20 Series:

  * Triggering firmware recovery by pressing a hardware button or calling the respective API.

>   * Dynamic Global Domain Frequency Scaling (DGFS), optimizing USB suspend power consumption.

Reviewer 4: At least in the source code its called GDFS (Global Domain Frequency Scaling Service)

Author: @shanthanordic Shantha confirmed the current name, I'd keep this as is, for now

Reviewer 5: Discussed it with @sveinaga , he said we will use what is in source code. So @FrancescoSer  please change ot GDFS.

  * Drivers adapted for the following peripherals:

    * UARTE120
    * PWM120
    * SPIM120
    * EXMIF

  * UICR validation support.
  * MSPI EXMIF support.

* Power Management:

  * System-On-All-Idle power improvements.

>   * Thread/Matter power efficiency improvements.

Reviewer 6: This is not part of the tag release yet, would be part of 3.0.0. So for.now i would recommend to remove this line.

Added the following features as experimental:

* Bluetooth:

  * :ref:`Parallel scanning and initiating connection <bt_scanning_while_connecting>`.
    It reduces the overall waiting time when there are several Bluetooth devices to be discovered and connected at the same time.

Improved the following features:

* Better BICR handling and generation.
* Enhanced power management features across various subsystems.

Implemented the following bug fixes:

* NCSDK-30802: Resolved an issue where the nRF54H20 device suddenly stopped transmitting ESB packets after nrfxlib 3.9.0.
* NCSDK-30161: Fixed an assertion during boot time caused by a combination of :kconfig:option:`CONFIG_ASSERT`, :kconfig:option:`CONFIG_SOC_NRF54H20_GPD`, and external flash.
* NCSDK-30117: Ensured that declaring a MEM component pointing to a memory region not assigned to a particular core is no longer possible.
* NCSDK-29682: Added support for the `cose-alg-sha-512` algorithm in the SUIT module.

Limitations
***********

> On the nRF54H20 SoC, The device firmware update (DFU) procedure from external flash memory does not work with the new flash memory driver for XIP.

Reviewer 7: The driver is not "for XIP". It uses the MSPI EXMIF driver to communicate with the flash chip, so I think this should be something like "... the new flash memory driver based on the MSPI EXMIF handling".

> On the nRF54H20 SoC, The device firmware update (DFU) procedure from external flash memory does not work with the new flash memory driver for XIP.

Reviewer 2: ```suggestion
On the nRF54H20 SoC, the Device Firmware Update (DFU) procedure from external flash memory does not work with the new flash memory driver for XIP.
```

Release tag
***********

The release tag for the |NCS| manifest repository (|ncs_repo|) is **v2.9.0-nRF54H20**.
Check the :file:`west.yml` file for the corresponding tags in the project repositories.

To use this release, check out the tag in the manifest repository and run ``west update``.
See :ref:`cloning_the_repositories` and :ref:`gs_updating_repos_examples` for more information.

> For information on the included repositories and revisions, see `Repositories and revisions for v2.9.0-nRF54H20`_.

Reviewer 2: The link.txt should be updated with `Repositories and revisions for v2.9.0-nRF54H20`_.

Integration test results
************************

The integration test results for this tag can be found in the following external artifactory:

* `Twister test report for nRF Connect SDK v2.9.0-nRF54H20`_

> * `Hardware test report for nRF Connect SDK v2.9.0-nRF54H20`_

Reviewer 2: link.txt should be updated with these link. Please confirm with "Stilwell, Thomas" if the link will be available for nRF54H20-RC1.

Author: Hi @thst-nordic, could you please confirm this?

Reviewer 5: @DatGizmo  should give this link

IDE and tool support
********************

`nRF Connect extension for Visual Studio Code <nRF Connect for Visual Studio Code_>`_ is the recommended IDE for |NCS| v2.9.0-nrf54h20-rc1.
See the :ref:`installation` section for more information about supported operating systems and toolchain.

Known issues
************

Known issues are only tracked for the latest official release.
See `known issues for nRF Connect SDK v2.9.0-nRF54H20`_ for the list of issues valid for the latest release.

Migration notes
***************

See the `Migration guide for nRF Connect SDK v2.9.0-nRF54H20`_ for the changes required or recommended when migrating your nRF54H Series application from |NCS| v2.8.0 to |NCS| v2.9.0-nrf54h20-rc1.

.. _ncs_release_notes_290-nRF54H20-rc1_changelog:

Changelog

> *********

Reviewer 2: Shouldn't we add these entries of nRF54H20? - https://github.com/nrfconnect/sdk-nrf/pull/19596/files#diff-67f56811be17161268a1ce745ea8aab07d9248a3ecc14987c87685b52c1e4cde, which are there in the 2.9 release notes

Author: Certain entries in common with 54H and other SoC should have been duplicated in the 2.9.0 release notes, without removing them from the changelog . If they got removed, it is easy to lose track of these items. I'll doublecheck and add them if missing

    * The module now waits until a triggered Bluetooth LE connection parameters update is completed before triggering subsequent updates for a given connection.
    * Improved the log to also display the information if USB is suspended.
      The information is needed to determine the requested connection parameters.
    * The module now uses non-zero Bluetooth LE peripheral latency while USB is suspended.
      This is done to prevent peripheral latency increase requests from :ref:`nrf_desktop_ble_latency` on peripheral's end.
    * The module reverts the USB suspended Bluetooth LE connection parameter update when USB cable is disconnected.

  * The :ref:`nrf_desktop_ble_scan` to always use a connection interval of 10 ms for peripherals without Low Latency Packet Mode (LLPM) support if a dongle supports LLPM and more than one Bluetooth LE connection.
    This is required to avoid Bluetooth Link Layer scheduling conflicts that could lead to HID report rate drop.

>   * The :ref:`suit_recovery` by adding support for triggering firmware recovery by pressing an hardware button or calling a dedicated API.

Reviewer 8: ```suggestion
  * The :ref:`suit_recovery` by adding support for triggering firmware recovery by pressing a hardware button or calling a dedicated API.
```

* :ref:`bt_fast_pair_readme` library:

  * Added support in build system for devices that do not support the :ref:`partition_manager`.
    The :ref:`zephyr:nrf54h20dk_nrf54h20` board target is the only example of such a device.
  * Updated the :c:func:`bt_fast_pair_info_cb_register` API to allow registration of multiple callbacks.

nRF RPC libraries
-----------------

* Added the :ref:`nrf_rpc_dev_info` library for obtaining information about a device connected through the :ref:`nrfxlib:nrf_rpc`.

sdk-nrfxlib
-----------

See the changelog for each library in the :doc:`nrfxlib documentation <nrfxlib:README>` for additional information.

Integrations
============

This section provides detailed lists of changes by :ref:`integration <integrations>`.

Google Fast Pair integration
----------------------------

* Added instructions on how to provision the Fast Pair data onto devices without the :ref:`partition_manager` support, specifically for the :ref:`zephyr:nrf54h20dk_nrf54h20`.

Zephyr
======

.. NOTE TO MAINTAINERS: All the Zephyr commits in the below git commands must be handled specially after each upmerge and each nRF Connect SDK release.

> The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``beb733919d8d64a778a11bd5e7d5cbe5ae27b8ee``, with some |NCS| specific additions.

Author: @shanthanordic Is this the right commit hash?

.. _ncs_release_notes_290-nrf54h20-rc1:

|NCS| v2.9.0-nRF54H20-rc1 Release Notes
#######################################

.. contents::
   :local:
   :depth: 2

|NCS| delivers reference software and supporting libraries for developing low-power wireless applications with Nordic Semiconductor products.
The SDK includes open source projects (TF-M, MCUboot, OpenThread, Matter, and the Zephyr RTOS), which are continuously integrated and redistributed with the SDK.

> The |NCS| v2.9.0-nrf54h20-rc1 is a nRF54H20-exclusive release tag, only supporting products based on the nRF54H20 SoC.

Reviewer 8: ```suggestion
The |NCS| v2.9.0-nrf54h20-rc1 is an nRF54H20-exclusive release tag, only supporting products based on the nRF54H20 SoC.
```

Release notes might refer to "experimental" support for features, which indicates that the feature is incomplete in functionality or verification, and can be expected to change in future releases.
To learn more, see :ref:`software_maturity`.

Highlights
**********

Added the following features as supported:

* nRF54H20 Series:

  * Triggering firmware recovery by pressing a hardware button or calling the respective API.
    For more information, see :ref:`ug_nrf54h20_suit_recovery`.
  * Dynamic Global Domain Frequency Scaling (DGFS), optimizing USB suspend power consumption.
  * Drivers adapted for the following peripherals:

>     * UARTE120

Reviewer 9: Timers as well

Author: @kl-cruz Could you list them?

    * PWM120
    * SPIM120
    * EXMIF

  * UICR validation support.
  * MSPI EXMIF support.

Added the following features as experimental:

> * Bluetooth:

Reviewer 8: ```suggestion
* Bluetooth®:
```

nRF Desktop
-----------

* Updated:

  * The :ref:`nrf_desktop_settings_loader` to make the :ref:`Zephyr Memory Storage (ZMS) <zephyr:zms_api>` the default settings backend for all board targets that use the MRAM technology.
    As a result, all :ref:`zephyr:nrf54h20dk_nrf54h20` configurations were migrated from the NVS settings backend to the ZMS settings backend.
  * :ref:`nrf_desktop_watchdog` by adding the :ref:`zephyr:nrf54h20dk_nrf54h20` release configuration.
  * The configuration files of the :ref:`nrf_desktop_click_detector` (:file:`click_detector_def.h`) to allow using them also when Bluetooth LE peer control using a dedicated button (:ref:`CONFIG_DESKTOP_BLE_PEER_CONTROL <config_desktop_app_options>`) is disabled.
  * The DTS description for board targets with a different DTS overlay file for each build type to isolate the common configuration that is now defined in the :file:`app_common.dtsi` file.
    The :ref:`zephyr:nrf54h20dk_nrf54h20` board configuration has been updated.

>   * The :ref:`nrf_desktop_ble_conn_params` to:

Reviewer 8: ```suggestion
  * The :ref:`nrf_desktop_ble_conn_params` with the following changes:
```

>     * Fix the Bluetooth LE connection parameters update loop (NCSDK-30261) that replicated if an nRF Desktop dongle without Low Latency Packet Mode (LLPM) support was connected to an nRF Desktop peripheral with LLPM support.

Reviewer 8: ```suggestion
    * Fixed the Bluetooth LE connection parameters update loop (NCSDK-30261) that replicated if an nRF Desktop dongle without Low Latency Packet Mode (LLPM) support was connected to an nRF Desktop peripheral with LLPM support.
```

>     * Wait until a triggered Bluetooth LE connection parameters update is completed before triggering subsequent updates for a given connection.

Reviewer 8: ```suggestion
    * The module now waits until a triggered Bluetooth LE connection parameters update is completed before triggering subsequent updates for a given connection.
```

>     * Improve the log to also display the information if USB is suspended.

Reviewer 8: ```suggestion
    * Improved the log to also display the information if USB is suspended.
```

      The information is needed to determine the requested connection parameters.

>     * Use non-zero Bluetooth LE peripheral latency while USB is suspended.

Reviewer 8: ```suggestion
    * The module now uses non-zero Bluetooth LE peripheral latency while USB is suspended.
```

      This is done to prevent peripheral latency increase requests from :ref:`nrf_desktop_ble_latency` on peripheral's end.

>     * Revert the USB suspended Bluetooth LE connection parameter update when USB cable is disconnected.

Reviewer 8: ```suggestion
    * The module reverts the USB suspended Bluetooth LE connection parameter update when USB cable is disconnected.
```

  * The :ref:`nrf_desktop_ble_scan` to always use a connection interval of 10 ms for peripherals without Low Latency Packet Mode (LLPM) support if a dongle supports LLPM and more than one Bluetooth LE connection.
    This is required to avoid Bluetooth Link Layer scheduling conflicts that could lead to HID report rate drop.

>   * The :ref:`suit_recovery` to:
> 
>     * Add support for triggering firmware recovery by pressing an hardware button or calling a dedicated API.
>       For more information, see :ref:`ug_nrf54h20_suit_recovery`.

Reviewer 8: ```suggestion
  * The :ref:`suit_recovery` by adding support for triggering firmware recovery by pressing an hardware button or calling a dedicated API.
    For more information, see :ref:`ug_nrf54h20_suit_recovery`.
```

    They contain several fixes related to malfunctioning in a heavy traffic environment and more.
    For details, see :ref:`zboss_changelog`.
  * The :ref:`ZBOSS Network Co-processor Host <ug_zigbee_tools_ncp_host>` package to the new version v2.2.5.

Applications
============

This section provides detailed lists of changes by :ref:`application <applications>`.

Machine learning
----------------

* Updated the application to enable the :ref:`Zephyr Memory Storage (ZMS) <zephyr:zms_api>` file system for the :ref:`zephyr:nrf54h20dk_nrf54h20` board.

IPC radio firmware
------------------

* Updated the application to enable the :ref:`Zephyr Memory Storage (ZMS) <zephyr:zms_api>` file system in all devices that contain MRAM, such as the nRF54H Series devices.

Matter bridge
-------------

* Added:

  * Support for the ``UniqueID`` attribute in the Bridged Device Basic Information cluster.
  * Version 2 of the bridged device data scheme containing ``UniqueID``.
  * Kconfig options :ref:`CONFIG_BRIDGE_MIGRATE_PRE_2_7_0 <CONFIG_BRIDGE_MIGRATE_PRE_2_7_0>` and :ref:`CONFIG_BRIDGE_MIGRATE_VERSION_1 <CONFIG_BRIDGE_MIGRATE_VERSION_1>` to enable migration from older data schemes.

> nRF Desktop

Reviewer 10: this part will be reviewed by @MarekPieta

Reviewer 11: A general note, I wonder a bit if this release notes shouldn't describe changes applied since NCS 2.9?

Author: @Marek As this release is not supposed to be used coming from 2.9, but the path for 54h users would be 2.8 -> 2.9.0-nRF54H20-rc1, I don't think this would be ideal.

-----------

* Updated:

  * The :ref:`nrf_desktop_settings_loader` to make the :ref:`Zephyr Memory Storage (ZMS) <zephyr:zms_api>` the default settings backend for all board targets that use the MRAM technology.
    As a result, all :ref:`zephyr:nrf54h20dk_nrf54h20` configurations were migrated from the NVS settings backend to the ZMS settings backend.
  * :ref:`nrf_desktop_watchdog` by adding the :ref:`zephyr:nrf54h20dk_nrf54h20` release configuration.
  * Updated the configuration files of the :ref:`nrf_desktop_click_detector` (:file:`click_detector_def.h`) to allow them to be used even when Bluetooth LE peer control using a dedicated button (:ref:`CONFIG_DESKTOP_BLE_PEER_CONTROL <config_desktop_app_options>`) is disabled.
  * The DTS description for board targets with a different DTS overlay file for each build type to isolate the common configuration that is now defined in the :file:`app_common.dtsi` file.
    The :ref:`zephyr:nrf54h20dk_nrf54h20` board configuration has been updated.

>   * The :ref:`nrf_desktop_ble_conn_params` with the following changes:

Reviewer 11: nRF54H20 DK does not use `nrf_desktop_ble_conn_params` module. No support for dongle (Bluetooth Central role)

Author: Will remove this

    * Fixed the Bluetooth LE connection parameters update loop (NCSDK-30261) that replicated if an nRF Desktop dongle without Low Latency Packet Mode (LLPM) support was connected to an nRF Desktop peripheral with LLPM support.
    * The module now waits until a triggered Bluetooth LE connection parameters update is completed before triggering subsequent updates for a given connection.
    * Improved the log to also display the information if USB is suspended.
      The information is needed to determine the requested connection parameters.
    * The module now uses non-zero Bluetooth LE peripheral latency while USB is suspended.
      This is done to prevent peripheral latency increase requests from :ref:`nrf_desktop_ble_latency` on peripheral's end.
    * The module reverts the USB suspended Bluetooth LE connection parameter update when USB cable is disconnected.

>   * The :ref:`nrf_desktop_ble_scan` to always use a connection interval of 10 ms for peripherals without Low Latency Packet Mode (LLPM) support if a dongle supports LLPM and more than one Bluetooth LE connection.

Reviewer 11: nRF54H20 DK does not use `nrf_desktop_ble_conn_params` module. No support for dongle (Bluetooth Central role)

Author: will remove this

    This is required to avoid Bluetooth Link Layer scheduling conflicts that could lead to HID report rate drop.

> nRF SoC flash driver

Reviewer 10: why is it a dedicated section? This should have been placed under the nRF Desktop category

Author: Will fix this

--------------------

> * Removed the ``imply`` for the partial erase feature of the nRF SoC flash driver (:kconfig:option:`CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE`) for the USB next stack (:ref:`CONFIG_DESKTOP_USB_STACK_NEXT <config_desktop_app_options>`).

Reviewer 11: nRF54H20 DK does not use `CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE` -> the board uses MRAM

Author: will remove this

  * Added support for the :ref:`zephyr:nrf54h20dk_nrf54h20` board target.

Edge Impulse samples
--------------------

* Added support for the :ref:`zephyr:nrf54h20dk_nrf54h20` board target in the following samples:

  * :ref:`ei_data_forwarder_sample`
  * :ref:`ei_wrapper_sample`

Matter samples
--------------

* Updated:

  * All Matter samples that support low-power mode to enable the :ref:`lib_ram_pwrdn` feature.
    It is enabled by default for the release configuration of the following samples:

    * :ref:`matter_lock_sample`
    * :ref:`matter_light_switch_sample`
    * :ref:`matter_smoke_co_alarm_sample`
    * :ref:`matter_window_covering_sample`

  * All Matter samples to enable the ZMS file subsystem in all devices that contain MRAM, such as the nRF54H Series devices.

* Disabled pausing Matter watchdog while CPU is in idle state in all Matter samples.
  To enable it, set the :ref:`CONFIG_NCS_SAMPLE_MATTER_WATCHDOG_PAUSE_IN_SLEEP<CONFIG_NCS_SAMPLE_MATTER_WATCHDOG_PAUSE_IN_SLEEP>` Kconfig option to ``y``.

* :ref:`matter_smoke_co_alarm_sample` sample:

  * Added support for ICD dynamic SIT LIT switching (DSLS).

SUIT samples
------------

> * Updated the :ref:`suit_recovery` by adding support for triggering firmware recovery by pressing a hardware button or calling a dedicated API.
>     For more information, see :ref:`ug_nrf54h20_suit_recovery`.

Reviewer 12: ```suggestion
* Updated the :ref:`suit_recovery` by adding support for triggering firmware recovery by pressing a hardware button or calling a dedicated API.
  For more information, see :ref:`ug_nrf54h20_suit_recovery`.
```

> * Updated the :ref:`suit_recovery` by adding support for triggering firmware recovery by pressing a hardware button or calling a dedicated API.
>     For more information, see :ref:`ug_nrf54h20_suit_recovery`.

Reviewer 2: ```suggestion
* Updated the :ref:`suit_recovery` by adding support for triggering firmware recovery by pressing a hardware button or calling a dedicated API.
  For more information, see :ref:`ug_nrf54h20_suit_recovery`.
```

The Zephyr fork in |NCS| (``sdk-zephyr``) contains all commits from the upstream Zephyr repository up to and including ``beb733919d8d64a778a11bd5e7d5cbe5ae27b8ee``, with some |NCS| specific additions.

For the list of upstream Zephyr commits (not including cherry-picked commits) incorporated into nRF Connect SDK since the most recent release, run the following command from the :file:`ncs/zephyr` repository (after running ``west update``):

.. code-block:: none

   git log --oneline beb733919d ^ea02b93eea

For the list of |NCS| specific commits, including commits cherry-picked from upstream, run:

.. code-block:: none

   git log --oneline manifest-rev ^beb733919d

The current |NCS| main branch is based on revision ``beb733919d`` of Zephyr.

.. note::
   For possible breaking changes and changes between the latest Zephyr release and the current Zephyr version, refer to the :ref:`Zephyr release notes <zephyr_release_notes>`.

Documentation
=============

* Added:

  * The :ref:`matter_samples_config` page that documents Kconfig options and snippets shared by Matter samples and applications.
  * A page about :ref:`add_new_driver`.
  * A page for the :ref:`sdp_gpio` application.

> 

Reviewer 12: Remove extra empty lines.
